lib/sdr_client/redesigned_client.rb

```diff
@@ -0,0 +1,162 @@
+# frozen_string_literal: true
+
+require 'logger'
+require 'singleton'
+
+module SdrClient
+  # The SDR client reimagined, built using patterns successfully used in other client gems we maintain
+  class RedesignedClient
```

Comment:
Can we shorten the new module name to, maybe SdrClient::Singleton or SdrClient2024?

Or consider making a new gem and retiring the old.  New name suggestion:  SdrApiClient.  Other suggestions welcome

Reply:
Reply from issue: https://github.com/sul-dlss/sdr-client/issues/261#issuecomment-1894570499